src/guide/customization.rs

```diff
@@ -0,0 +1,398 @@
+//! }
+//!
+//! impl eyre::EyreContext for CustomContext {
+//!     fn default(inner: &(dyn std::error::Error + 'static)) -> Self {
+//!         let backtrace = if Chain::new(source).all(|e| e.backtrace().is_none()) {
+//!             Some(Backtrace::capture())
+//!         } else {
+//!             None
+//!         };
+//!
+//!         Self {
+//!             // set a default status
+//!             status: StatusCode::INTERNAL_SERVER_ERROR,
+//!             backtrace
+//!         }
+//!     }
+//!
+//!     fn debug(
+//!         &self,
+//!         error: &(dyn std::error::Error + 'static),
+//!         f: &mut core::fmt::Formatter<'_>,
+//!     ) -> core::fmt::Result {
+//!         if f.alternate() {
+//!             return core::fmt::Debug::fmt(error, f);
+//!         }
+//!
+//!         // print our status on the first line
+//!         write!(f, "{}", self.status)?;
+//!
+//!         let errors = Chain::new(error).enumerate();
+//!
+//!         for (n, error) in errors {
+//!             writeln!(f)?;
+//!             write!(Indented::numbered(f, n), "{}", error)?;
+//!         }
+//!
+//!         let backtrace = self
+//!             .backtrace
+//!             .as_ref()
+//!             .or_else(|| Chain::new(error).rev().filter_map(|e| e.backtrace()).next())
+//!             .expect("expected: backtrace is always captured");
+//!
+//!         if let BacktraceStatus::Captured = backtrace.status() {
+//!             write!(f, "\n\nStack backtrace:\n{}", self.backtrace)?;
+//!         }
+//!
+//!         Ok(())
+//!     }
+//! }
+//! ```
+//!
+//! This is great all, but now we just get INTERNAL_SERVER_ERROR every time, which isn't very
+//! useful, lets add a helper trait of our own, similar to the provided `WrapErr` trait, for
+//! setting the status.
+//!
+//! ```rust
+//! pub(crate) trait Status {
+//!     type Result;
+//!     fn set_status(self, status: StatusCode) -> Self::Result;
+//! }
+//!
+//! impl<T, E> Status for Result<T, E>
+//! where
+//!     E: Into<Report>,
+//! {
+//!     type Result = Result<T, Report>;
+//!
+//!     fn set_status(self, status: StatusCode) -> Self::Result {
+//!         self.map_err(|e| {
+//!             let mut reporter = e.into();
+//!             reporter.context_mut().status = status;
+//!             reporter
+//!         })
+//!     }
+//! }
+//! ```
+//!
+//! Now we can use our reporter to associate status codes with any error we throw with
+//! INTERNAL_SERVER_ERROR as a hopefully reasonable default should we forget / not want to set the
+//! status. Here's what our error handling api ends up looking like in practice:
+//!
+//! ```rust
+//! pub(crate) async fn get(&self, timeframe: Timeframe) -> Result<Vec<u8>, Report> {
+//!     let mut map = self.map.lock().unwrap();
+//!     let entry = map
+//!         .get_mut(&timeframe)
+//!         .ok_or_else(|| eyre!("No entries for this timestamp"))
+//!         .set_status(StatusCode::NOT_FOUND)?;
+//!
+//!     // ...
+//! }
+//! ```
+//!
```

Comment:
I think including some example code here which reads the status from an error as well would be awesome!